posthog/api/event.py

```diff
@@ -100,13 +100,22 @@ class EventViewSet(StructuredViewSetMixin, viewsets.ModelViewSet):
 
     def get_queryset(self):
         queryset = cast(EventManager, super().get_queryset()).add_person_id(self.team_id)
-
         if self.action == "list" or self.action == "sessions" or self.action == "actions":
             queryset = self._filter_request(self.request, queryset)
 
         order_by_param = self.request.GET.get("orderBy")
         order_by = ["-timestamp"] if not order_by_param else list(json.loads(order_by_param))
-        return queryset.order_by(*order_by)
+        queryset = queryset.order_by(*order_by)
+        limit_raw = self.request.GET.get("limit")
+        limit: Optional[int]
+        if limit_raw:
+            try:
+                limit = int(limit_raw)
+                valid_ids = queryset.values_list("pk", flat=True)[:limit]
+                queryset = queryset.filter(pk__in=valid_ids)
+            except ValueError:
+                raise exceptions.ValidationError("Query param limit must be omitted or an integer!")
```

Comment:
It's best practice for the try statement to only include the relevant line, in this case parsing to int. As a `ValueError` raised in statements on lines 114 & 115 could be a different problem. You could also try doing this with a single var?